packages/shared/scope-hoisting/src/formats/esmodule.js

```diff
@@ -170,17 +207,22 @@ export function generateExports(
         areArraysStrictlyEqual(ids, exportedSymbolsBindings) &&
         !path.isImportDeclaration()
       ) {
-        path.replaceWith(t.exportNamedDeclaration(path.node, []));
+        // We don't update the references in `node` itself (e.g. init), because this statement
+        // will never be removed and therefore the shaking doesn't need correct
+        // information. All existing references in `node` are "dead" but will also never be removed.
```

Comment:
This is quite unfortunate regarding scope correctness assertions in development.